### ISSUE TITLE
MenuItem click handler not executed

### ISSUE THREAD

**Issue by Moai-** [electron/electron#8039]
* Electron version: 1.3.9
* Operating system: Windows 10

### Expected behavior

When a menu item is created with a `click` function, that function should be executed when the item is clicked.

### Actual behavior

The `click` function is ignored.

### How to reproduce

Unfortunately I can't find a simple way to reproduce this bug. My program does use multiple Electron `browserWindow`s, as well as Angular.js as a framework. The problematic menus occur inside included Angular directives. However, no matter how I try, I can't reproduce this issue outside of my specific app. The worst part is that it was working last night, and stopped working at an unknown point between now and then, without a commit in between.

Are there any known situations in which the `click` handler isn't executed at all? Everything else about the menu works. Creating simple menus just using the `role` options works, and the menus execute just like they should. The menus pop up, and are otherwise formatted exactly how I put them together. The only problem is the `click` not running at all.

This is my example code, included inside of a `link` function of an Angular directive:

```
$element.on('contextmenu', e=>{
    let menu = new Menu();
    menu.append(new MenuItem({
      label: 'test',
      click() {
        console.log('test');
       }
    }));
    menu.popup(remote.getCurrentWindow());
})
```

**Comment by Moai-** [comment]
I have resolved my issue, will post solution in case if anyone else runs into it. It's kind of a weird one.

As previously mentioned, my app has multiple windows. These new windows are spawned when the user "undocks" one of the in-app tabs. I use IPC for coordinating shared data between these tabs.

On exit, every window needs to clean up its IPC listeners, otherwise the listeners stack up and the app leaks memory. So on exit, I used the following code:

`ipc.removeAllListeners();`

Apparently Menus are somehow related to IPC, because this is the line that wipes out the `click` handlers from any Menus in the app. Thankfully, my coordination system only uses one channel, so my issue was solved thusly:

`ipc.removeAllListeners('update-in');`

I'm sure this isn't a bug but a really weird design feature, but I will leave this issue open anyway, in case if it does turn out to be a bug, or if the devs would like to document this behaviour (unless it's already documented and I'm just blind). Either way, this was a very weird issue.
